zcash_client_backend/src/tor.rs

```diff
@@ -0,0 +1,104 @@
+//! Tor support for Zcash wallets.
+
+use std::{fmt, io, path::Path};
+
+use arti_client::{config::TorClientConfigBuilder, TorClient};
+use tor_rtcompat::PreferredRuntime;
+use tracing::debug;
+
+mod http;
+
+/// A Tor client that exposes capabilities designed for Zcash wallets.
+pub struct Client {
+    inner: TorClient<PreferredRuntime>,
+}
+
+impl Client {
+    /// Creates and bootstraps a Tor client.
+    ///
+    /// The client's persistent data and cache are both stored in the given directory.
+    /// Preserving the contents of this directory will speed up subsequent calls to
+    /// `Client::create`.
+    ///
+    /// Returns an error if `tor_dir` does not exist, or if bootstrapping fails.
+    pub async fn create(tor_dir: &Path) -> Result<Self, Error> {
+        let runtime = PreferredRuntime::current()?;
+
+        if !tokio::fs::try_exists(tor_dir).await? {
+            return Err(Error::MissingTorDirectory);
+        }
+
+        let mut config_builder = TorClientConfigBuilder::from_directories(
+            tor_dir.join("arti-data"),
+            tor_dir.join("arti-cache"),
+        );
+        config_builder
+            .storage()
+            .permissions()
+            .dangerously_trust_everyone();
```

Comment:
Can you leave a comment explaining why this option is chosen?

Reply:
Oops, this _was_ commented on, but in a PR comment on the Swift FFI PR before I extracted the changes into this PR. https://github.com/Electric-Coin-Company/zcash-light-client-ffi/pull/142#discussion_r1628700953
> The permissions here need testing. I was having trouble getting Arti to work in the iOS simulator, and one of the things I did to get around it was enable this setting. I eventually also figured out that on the Swift side we needed to pass over `URL.path` instead of `URL.absoluteString`, but I haven't gone back and experimented with other permissions here yet.

Reply:
I've now tested this branch on both Android and iOS without this setting, and they both appear to work. I'll remove it.